incubator/quarkus/image/Dockerfile-stack

```diff
@@ -1,16 +1,17 @@
 # Dockerfile for building the stack
-FROM quay.io/quarkus/centos-quarkus-maven:19.2.1
+FROM quay.io/quarkus/centos-quarkus-maven:19.3.1-java8
 
 COPY ./LICENSE /licenses/
 COPY ./project /project
 COPY ./config /config
 
 WORKDIR /project
+RUN mkdir "user-app"
 
 RUN mvn -N io.takari:maven:wrapper -Dmaven=$(mvn help:evaluate -Dexpression=maven.version -q -DforceStdout)
 RUN mvn -B install dependency:go-offline -DskipTests
 
-ENV APPSODY_MOUNTS=".:/project/user-app"
+ENV APPSODY_MOUNTS="src:/project/user-app/src;pom.xml:/project/user-app/pom.xml;mvnw:/project/user-app/mvnw;mvnw.cmd:/project/user-app/mvnw.cmd"
```

Comment:
Can we avoid mounting individual files? This is not officially supported by Docker on Windows and has caused us a lot of grief on other stacks.

Reply:
I've been looking into this and suggested this solution. The thing is that maven can't create `target` directory inside of mounted directory. One solution might be creating `target` directory  with `rwx` rights for `others` when initializing project, but I don't know if that can be easily. Another solution might be mounting `target` as persistent volume similarly to `"/home/quarkus/.m2"`. Question is: does `APPSODY_DEPS` support multiple directories? If so maybe adding `/project/user-app/target` to `APPSODY_DEPS` could help.

Reply:
@matejvasek - There's an open issue targeting this milestone to support this: https://github.com/appsody/appsody/issues/934

Reply:
We will aim to support for multiple volumes in next release. However, you could potentially create the folder with write permission in the template if you want.

Reply:
I tried creating `target` directory with `0777` mod in template. Unfortunately it wouldn't work.
a) `appsody stack package` internally uses `cp -rf` which drops write access for `others`.
That might be fixed by `cp -af`.
b) Even if the a) is fixed `initUntar` at `init.go:406` disregards mod in tarball and uses `0755` for directories.
c) Even if both a) and b) are fixed syscall mkdir("a-dir-name", 0777) actually creates directory with mod `0775`, probably due system `umask` setting.

Is there any possibility of running some kind of post init script (which could call `chmod`)? I know there is `.appsody-init.sh`, but it's called inside of a container, I need to run post init script locally.

Reply:
@paulrobinson I tried the development [branch](https://github.com/skoh7645/appsody/tree/trackproject-volumes) for multi dep. mounts. It seems to work, it's only necessary to add `RUN mkdir -p /project/user-app/target` into `image/Dockerfile-stack`.

Reply:
Thanks @matejvasek. I'll see if I can find the time to give it a go tomorrow.

Reply:
> I tried creating `target` directory with `0777` mod in template. Unfortunately it wouldn't work.
> a) `appsody stack package` internally uses `cp -rf` which drops write access for `others`.
> That might be fixed by `cp -af`.
> b) Even if the a) is fixed `initUntar` at `init.go:406` disregards mod in tarball and uses `0755` for directories.
> c) Even if both a) and b) are fixed syscall mkdir("a-dir-name", 0777) actually creates directory with mod `0775`, probably due system `umask` setting.
> 
> Is there any possibility of running some kind of post init script (which could call `chmod`)? I know there is `.appsody-init.sh`, but it's called inside of a container, I need to run post init script locally.

The appsody-init.sh scripts do actually run locally on the host.

Reply:
Would we want to get this in before we have support for multiple mounts? I was thinking that the target problem is a common issue with all java/maven stacks.

Reply:
@neeraj-laad ohh you are right about `.appsody-init.sh`.
Adding
```shell
mkdir -p ../target
chmod o+w ../target
```
Seems to help.

Reply:
When investigating this I noticed two solutions that existing java stack use now:

a) `ENV APPSODY_USER_RUN_AS_LOCAL=true` this is used by `incubator/java-spring-boot-2`, with that option there is no problem with creating `target` directory. It works for the `incubator/java-spring-boot-2` stack because the stack also mounts local maven repository too (so all access right are ok).

`experimental/quarkus` doesn't mount maven repository (`~/.m2/repository`) from local user -- it is created when building an image by `image/Dockerfile-stack` and it is created under `quarkus` user. So If we used `ENV APPSODY_USER_RUN_AS_LOCAL=true`  we wouldn't have a problem with creating `target` directory, but we would have a problem with accessing mvn repo.

b) mounting individual files/directories -- this is used by `incubator/java-microprofile`, but this won't work for Windows.

Reply:
Would we like "`ENV APPSODY_USER_RUN_AS_LOCAL=true` + mounting local mvn repo" solution more?

Reply:
@matejvasek, I assume I need to wait for https://github.com/skoh7645/appsody/tree/trackproject-volumes to make it into a Appsody release before I can have this PR merged? Do you know the timeframes for that?

Reply:
The APPSODY_USER_RUN_AS_LOCAL and using mvn cache should be independent. I think you can still use a volume if you run as local. I prefer not mounting anything from user's m2 cache as that is a much cleaner way to using containers.